src/link.c

```diff
@@ -607,14 +614,15 @@ static void on_session_flow_on(void* context)
 static void on_send_complete(void* context, IO_SEND_RESULT send_result)
 {
     LIST_ITEM_HANDLE delivery_instance_list_item = (LIST_ITEM_HANDLE)context;
-    DELIVERY_INSTANCE* delivery_instance = (DELIVERY_INSTANCE*)singlylinkedlist_item_get_value(delivery_instance_list_item);
-    LINK_INSTANCE* link_instance = (LINK_INSTANCE*)delivery_instance->link;
+    ASYNC_OPERATION_HANDLE pending_delivery_operation = (ASYNC_OPERATION_HANDLE)singlylinkedlist_item_get_value(delivery_instance_list_item);
+    DELIVERY_INSTANCE* delivery_instance = (DELIVERY_INSTANCE*)GET_ASYNC_OPERATION_CONTEXT(DELIVERY_INSTANCE, pending_delivery_operation);
```

Comment:
>DELIVERY_INSTANCE* delivery_instance = (DELIVERY_INSTANCE*) [](start = 4, length = 59)

I'm unfamiliar with the API, so forgive me for my ignorance.  Should you check for NULL here?

Reply:
Not needed as it is put in by the same module.